### DISCUSSION TITLE
how to change the order of the titles

### DISCUSSION THREAD

**Discussion by hongthiet1994** [Hieromon/AutoConnect#510] [How can I?] [answered]
![image](https://user-images.githubusercontent.com/29925688/181199982-30eac33c-c84b-4803-a4c0-dce72e047c91.png)

How to change order of this

**Comment by Hieromon** [comment]
What is `this`? If you don`t present clearly what you have done, it will be difficult for many people to help you.
Anyway, this kind of topic goes to Discussions.

**Comment by hongthiet1994** [comment]
how to change order of "HOME" "EEPROM" "Reset" "Time zone"  title

**Comment by Hieromon** [comment, marked as answer]
@hongthiet1994 If you are envisioning an API that allows you to easily change the order on the sketch, AutoConnect does not provide such an interface.
However, if you are willing to modify the AutoConnect source code to achieve this, the part that generates the menu HTML is as follows:
https://github.com/Hieromon/AutoConnect/blob/8f6076266f9faf71493b4e58b4eb43ea287bbc3e/src/AutoConnectPage.cpp#L970-L981

In the above *_token_MENU_PRE* function, it is the concatenation order of the String _menuItem_ variable that determines some menu orderliness. By changing this concatenation order, you can change the order of the **Config New** and **Open SSIDs** and **Disconnect** and **RESET** items.

On the other hand, changing the menu item for a custom web page that includes AutoConnectAux --- you see it as **Time/Zone**, **EEPROM** --- requires knowledge of PageBuilder. It is a so-called simple lexical analyzer with templates and tokens. You can find its structure in the [PageBuilder repository](https://github.com/Hieromon/PageBuilder). The declaration of the template in the second half part of the menu item is located at:
https://github.com/Hieromon/AutoConnect/blob/8f6076266f9faf71493b4e58b4eb43ea287bbc3e/src/AutoConnectAux.cpp#L29-L72

The code to replace the `{{MEMU_POST}}` token in the above template with HTML is the following part:
https://github.com/Hieromon/AutoConnect/blob/8f6076266f9faf71493b4e58b4eb43ea287bbc3e/src/AutoConnectPage.cpp#L983-L990

Note that the order of menu items to the custom web page depends on the order of loading. Its loading order matches the order defined in JSON or specified as a parameter list of the [AutoConnect::join](https://hieromon.github.io/AutoConnect/api.html#join) function.
